File: POBJ/Csharp/Ex_4_Reprise_Cpp/Ex_4_Reprise_Cpp/Utilisateur.cs
using System;

public class Utilisateur
{
    private int m_CodeFormeChoisie;
    private double m_x1;
    private double m_x2;
    private FormesGeometriques myFormesGeometriques;

    public void AfficherResultats()
    {
        if (m_CodeFormeChoisie < 3)
        {
            if (m_CodeFormeChoisie == 0)
            {
                // Affichage pour le TriangleRectangle
                Console.WriteLine($"Pour un TriangleRectangle base = {m_x1} hauteur = {m_x2} surface = {myFormesGeometriques.GetSurface()} perimetre = {myFormesGeometriques.GetPerimetre()}");
            }
            else if (m_CodeFormeChoisie == 1)
            {
                // Affichage pour le Rectangle
                Console.WriteLine($"Pour un Rectangle largeur = {m_x1} hauteur = {m_x2} surface = {myFormesGeometriques.GetSurface()} perimetre = {myFormesGeometriques.GetPerimetre()}");
            }
            else if (m_CodeFormeChoisie == 2)
            {
                // Affichage pour le Cercle
                Console.WriteLine($"Pour un Cercle diametre = {m_x1} surface = {myFormesGeometriques.GetSurface()} perimetre = {myFormesGeometriques.GetPerimetre()}");
            }
            else
            {
                // Gestion d'une valeur de m_CodeFormeChoisie non prise en charge
                Console.Error.WriteLine("Code de forme non pris en charge.");
            }
        }
        else
        {
            Console.Error.WriteLine("Erreur : Aucune forme choisie pour afficher les resultats.");
        }
    }

    public int ChoisirForme(int CodeForme)
    {
        m_CodeFormeChoisie = CodeForme;

        if (m_CodeFormeChoisie == 0)
        {
            myFormesGeometriques = new TriangleRectangle();
        }
        else if (m_CodeFormeChoisie == 1)
        {
            myFormesGeometriques = new Rectangle();
        }
        else if (m_CodeFormeChoisie == 2)
        {
            myFormesGeometriques = new Cercle();
        }
        return m_CodeFormeChoisie;
    }

    public void SaisirLesParametres()
    {
        if (m_CodeFormeChoisie == 0)
        {
            // Saisie des paramètres pour le TriangleRectangle
            Console.Write("Entrer la valeur de la base du TriangleRectangle : ");
            m_x1 = Convert.ToDouble(Console.ReadLine());

            Console.Write("Entrer la valeur de la hauteur du TriangleRectangle : ");
            m_x2 = Convert.ToDouble(Console.ReadLine());
        }
        else if (m_CodeFormeChoisie == 1)
        {
            // Saisie des paramètres pour le Rectangle
            Console.Write("Entrer la valeur de la largeur du Rectangle : ");
            m_x1 = Convert.ToDouble(Console.ReadLine());

            Console.Write("Entrer la valeur de la hauteur du Rectangle : ");
            m_x2 = Convert.ToDouble(Console.ReadLine());
        }
        else if (m_CodeFormeChoisie == 2)
        {
            // Saisie des paramètres pour le Cercle
            Console.Write("Entrer la valeur du diametre du Cercle : ");
            m_x1 = Convert.ToDouble(Console.ReadLine());
            m_x2 = 0;
        }
        else
        {
            // Gestion d'une valeur de m_CodeFormeChoisie non prise en charge
            Console.Error.WriteLine("Code de forme non pris en charge.");
        }
    }

    public void ExecuterLesCalculs()
    {
        // Vérifier si m_CodeFormeChoisie est valide
        if (m_CodeFormeChoisie < 3)
        {
            // Appeler les méthodes de calcul de la classe FormesGeometrique
            myFormesGeometriques.CalculerSurface(m_x1, m_x2);
            myFormesGeometriques.CalculerPerimetre(m_x1, m_x2);
        }
        else
        {
            Console.Error.WriteLine("Erreur : Aucune forme choisie pour effectuer les calculs.");
        }
    }
}

File: POBJ/CPP/Exercices/EX_4/EX_4/Utilisateur.cpp
#include "Utilisateur.h"
#include "TriangleRectangle.h"
#include "Cercle.h"
#include "Rectangle.h"
#include <iostream>




void Utilisateur::AfficherResultats()
{
    if (m_CodeFormeChoisie < 3)
    {
        if (m_CodeFormeChoisie == 0)
        {
            // Affichage pour le TriangleRectangle
            std::cout << "Pour un TriangleRectangle base = " << m_x1 << " hauteur = " << m_x2
                << " surface = " << myFormesGeometriques->GetSurface() << " perimetre = " << myFormesGeometriques->GetPerimetre() << std::endl;
        }
        else if (m_CodeFormeChoisie == 1)
        {
            // Affichage pour le Rectangle
            std::cout << "Pour un Rectangle largeur = " << m_x1 << " hauteur = " << m_x2
                << " surface = " << myFormesGeometriques->GetSurface() << " perimetre = " << myFormesGeometriques->GetPerimetre() << std::endl;
        }
        else if (m_CodeFormeChoisie == 2)
        {
            // Affichage pour le Cercle
            std::cout << "Pour un Cercle diametre = " << m_x1
                << " surface = " << myFormesGeometriques->GetSurface() << " perimetre = " << myFormesGeometriques->GetPerimetre() << std::endl;
        }
        else
        {
            // Gestion d'une valeur de m_CodeFormeChoisie non prise en charge
            std::cerr << "Code de forme non pris en charge." << std::endl;
        }
    }
    else
    {
        std::cerr << "Erreur : Aucune forme choisie pour afficher les resultats." << std::endl;
    }

    
}

int Utilisateur::ChoisirForme(int CodeForme)
{
    //{
    //    Cercle test = Cercle(); // Mémoire gérée automatiquement
    //}

    //{
    //    Cercle* test = new Cercle(); // Mémoire gérée manuellement
    //    delete test; // Je libère la zone mémoire pointée par le pointeur
    //}


    m_CodeFormeChoisie = CodeForme;

    if (m_CodeFormeChoisie == 0)
    {
        myFormesGeometriques = new TriangleRectangle();
    }
    else if (m_CodeFormeChoisie == 1)
    {
        myFormesGeometriques = new Rectangle();
    }
    else if (m_CodeFormeChoisie == 2)
    {
        myFormesGeometriques = new Cercle();
    }
    return m_CodeFormeChoisie;
}

void Utilisateur::SaisirLesParametres()
{
    if (m_CodeFormeChoisie == 0)
    {
        // Saisie des paramètres pour le TriangleRectangle
        std::cout << "Entrer la valeur de la base du TriangleRectangle : ";
        std::cin >> m_x1;

        std::cout << "Entrer la valeur de la hauteur du TriangleRectangle : ";
        std::cin >> m_x2;
    }
    else if (m_CodeFormeChoisie == 1)
    {
        // Saisie des paramètres pour le TriangleRectangle
        std::cout << "Entrer la valeur de la largeur du Rectangle : ";
        std::cin >> m_x1;

        std::cout << "Entrer la valeur de la hauteur du Rectangle : ";
        std::cin >> m_x2;
    }   
    else if (m_CodeFormeChoisie == 2)
    {
        // Saisie des paramètres pour le TriangleRectangle
        std::cout << "Entrer la valeur du diametre du Cercle : ";
        std::cin >> m_x1;
        m_x2 = 0;
    }
    else
    {
        // Gestion d'une valeur de m_CodeFormeChoisie non prise en charge
        std::cerr << "Code de forme non pris en charge." << std::endl;
    }
}

void Utilisateur::ExecuterLesCalculs()
{
    {
        // Vérifier si m_CodeFormeChoisie est valide
        if (m_CodeFormeChoisie < 3)
        {
            // Appeler les méthodes de calcul de la classe FormesGeometrique
             myFormesGeometriques->CalculerSurface(m_x1, m_x2);
             myFormesGeometriques->CalculerPerimetre(m_x1, m_x2);
        }
        else
        {
            std::cerr << "Erreur : Aucune forme choisie pour effectuer les calculs." << std::endl;
        }
    }


}

//Destructeur
Utilisateur::~Utilisateur()
{
    //std::cout << "Destructeur de Utilisateur " << myFormesGeometriques << std::endl;
    delete myFormesGeometriques; // Libérer la mémoire pointée par le pointeur
}
